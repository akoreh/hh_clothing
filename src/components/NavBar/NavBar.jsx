import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { selectCartDropdownHidden } from '../../redux/cart/cart.selectors';

import ShoppingBagIcon from '../ShoppingBagIcon/ShoppingBagIcon';
import CartDropdown from '../CartDropdown/CartDropdown';

import {NavBarContainer, OptionsContainer, OptionLink, Logo} from './NavBar.styles';

const NavBar = ({cartDropdownHidden}) => (
    <NavBarContainer>
        <Logo>HH</Logo>
        <OptionsContainer>
            <OptionLink to="/shop">SHOP</OptionLink>
            <OptionLink to="/shop">CONTACT</OptionLink>
            <OptionLink to="/shop">SIGN IN</OptionLink>
            <ShoppingBagIcon itemCount={0} />
        </OptionsContainer>
        {!cartDropdownHidden ? <CartDropdown /> : null}
    </NavBarContainer>
);

const mapStateToProps = createStructuredSelector({
    cartDropdownHidden: selectCartDropdownHidden
});

export default connect(mapStateToProps)(NavBar);