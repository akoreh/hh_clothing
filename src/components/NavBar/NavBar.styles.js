import styled from 'styled-components';
import { Link } from 'react-router-dom';

import {black} from '../../styles/colors';

const navBarHeight = 70;
const halfOfNavBarHeight = navBarHeight / 2;

export const NavBarContainer = styled.div`
  height: ${navBarHeight}px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-bottom: 25px;

  @media screen and (max-width: 800px) {
    height: 60px;
    padding: 10px;
    margin-bottom: 20px;
  }
`;

export const Logo = styled.h1`
  font-weight: 300;
  color: ${black};
  font-size: ${halfOfNavBarHeight}px;
  border-bottom: 1px solid ${black};
  border-top: 1px solid ${black};
  line-height: ${halfOfNavBarHeight}px;
  height: ${halfOfNavBarHeight}px;
  width: ${navBarHeight}px;
  text-align: center;
`;

export const OptionsContainer = styled.div`
  width: 50%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: flex-end;

  @media screen and (max-width: 800px) {
    width: 80%;
  }
`;

export const OptionLink = styled(Link)`
  padding: .8rem 1rem;
  cursor: pointer;
  font-size: 1.3rem;
`;

OptionLink.displayName = 'OptionLink';
