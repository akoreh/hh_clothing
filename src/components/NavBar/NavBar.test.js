import React from 'react';

import { shallow } from 'enzyme';

import NavBar from './NavBar';

describe('NavBar component', () => {
    it('should render NavBar', () => {
        const wrapper = shallow(<NavBar />);
        expect(wrapper).toMatchSnapshot();
    });
});