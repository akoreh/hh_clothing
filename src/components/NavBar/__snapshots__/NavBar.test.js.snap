// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render NavBar 1`] = `ShallowWrapper {}`;
